docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+
+    hamiltonian = PauliSumOp.from_list([('X', 1), ('Z',1)])
+    evolution = PauliTrotterEvolution(trotter_mode=Trotter(), reps=1)
+    evol_result = evolution.convert(hamiltonian.exp_i())
+    evolved_state = evol_result.to_circuit()
+.. raw:: html
+
+    </details>
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Alternative</b></a></summary>
+
+.. code-block:: python
+
+            from qiskit.quantum_info import SparsePauliOp
+            from qiskit.synthesis import SuzukiTrotter
+            from qiskit.circuit.library import PauliEvolutionGate
+            from qiskit import QuantumCircuit
+
+            hamiltonian = SparsePauliOp.from_list([('X', 1), ('Z',1)])
+            evol_gate = PauliEvolutionGate(hamiltonian, 1, synthesis=SuzukiTrotter())
+            evolved_state = QuantumCircuit(1)
+            evolved_state.append(evol_gate, [0])
+
+.. raw:: html
+
+    </details>
+
+
+Example 2: Matrix evolution
+############################
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Opflow</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit.opflow import MatrixEvolution, MatrixOp
+
+    hamiltonian = MatrixOp([[0, 1], [1, 0]])
+    evolution = MatrixEvolution()
+    evol_result = evolution.convert(hamiltonian.exp_i())
+    evolved_state = evol_result.to_circuit()
+
+.. raw:: html
+
+    </details>
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Alternative</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit.quantum_info import SparsePauliOp
+    from qiskit.extensions import HamiltonianGate
+    from qiskit import QuantumCircuit
+
+    evol_gate = HamiltonianGate([[0, 1], [1, 0]], 1)
+    evolved_state = QuantumCircuit(1)
+    evolved_state.append(evol_gate, [0])
+
+.. raw:: html
+
+    </details>
+
+Expectations
+------------
+*Back to* `Index`_
+
+Expectations are converters which enable the computation of the expectation value of an observable with respect to some state function.
+This functionality can now be found in the estimator primitive.
+
+Algorithm-Agnostic Expectations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*Back to* `Index`_
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.expectations.ExpectationFactory`
+     - No direct replacement. This class was used to create instances of one of the classes listed below.
+
+   * - :class:`~qiskit.opflow.expectations.AerPauliExpectation`
+     - Use :class:`~Estimator` primitive from ``qiskit_aer`` with ``approximation=True`` and ``shots=None`` as ``run_options``.
+       See example :ref:`below <expect_state>`.
+
+   * - :class:`~qiskit.opflow.expectations.MatrixExpectation`
+     - Use :class:`~Estimator` primitive from ``qiskit`` (if no shots are set, it performs an exact Statevector calculation).
+       See example :ref:`below <matrix_state>`.
+
+   * - :class:`~qiskit.opflow.expectations.PauliExpectation`
+     - Use any :class:`~Estimator` primitive from ``qiskit``.
```

Comment:
Here we have to be a bit careful and mention that shots have to be set or that users should use the Aer estimator 🙂